internal/beatcmd/reloader.go

```diff
@@ -126,10 +148,15 @@ func (r *Reloader) reloadInputs(configs []*reload.ConfigWithMeta) error {
 func (r *Reloader) reloadOutput(cfg *reload.ConfigWithMeta) error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
+	if r.outputConfig != nil && configEqual(cfg.Config, r.outputConfig) {
```

Comment:
Is the `configEqual` check strictly necessary for switching to Elastic Agent v2? In the past we consciously didn't introduce `reflect.*` in production code. In which cases would the config be reloaded without any change, do we expect performance improvement from this?

Reply:
This is necessary for working around https://github.com/elastic/elastic-agent/issues/1738

This is not a hot code path, so it won't matter. Nevertheless I intend to back this change out after the issue is fixed.